src/components/views/BlackCardSelection.js

```diff
@@ -3,35 +3,63 @@ import "styles/views/GameMenu.scss";
 import CardButton from "../ui/CardButton";
 import Header from "./Header";
 import {api, handleError} from "../../helpers/api";
-import {useHistory} from "react-router-dom";
+import {useHistory, useLocation} from "react-router-dom";
 import BaseContainer from "../ui/BaseContainer";
 
 
 
 const BlackCardSelection = () => {
 
-    const history = useHistory();
+    const history = useHistory()
     const [cards, setCards] = useState(null)
 
+    // Because of rendering reasons, we use location here, which allows passing states around components.
+    // Here we get the state from Registration / Login, because the localStorage might not have been updated yet.
+    // If token / userId is in localStorage, we use this, else we use the state passed from login / registration
+    // if we're coming from these components, else we use null and trigger a call to the server which we shall then handle.
+    const location = useLocation()
+    let id = null
+    let token = null
+    try {
+        id = (localStorage.getItem("id")) ? localStorage.getItem("id") : location.state.id
+    } catch (e) {
+        console.log("No userId found!")
+    }
+    try {
+        token = (localStorage.getItem("token")) ? localStorage.getItem("token") : location.state.token
+    } catch (e) {
+        console.log("No token found!")
+    }
+
+
+    // fetch the blackCards from the server (it is the server's responsibility to give us 8 cards)
     useEffect(() => {
         async function fetchCards() {
             try {
-                const response = await api.get(`games/${localStorage.getItem("id")}`);
+                const response = await api.get(`games/${id}`,
+                    {
+                        // reconfiguration might be necessary in case token is not in localStorage here
+                        headers: {
+                            "authorization": token
```

Comment:
shouldn't the token be passed anyways in the api.js file? Or is this something different

Reply:
not sure what you mean. having the auth token in the header of the request is standard practice.

Reply:
@JoMuelle Yes, but how we solved it introduced a bug. Will fix this in a seperate Issue
@TwoDigitsOneNumber We do not want to reconfigure the header each time we call api, so this is what he meant.